### PR TITLE
Limit Editing on Levelbuilder to only en-US Locale

### PR DESCRIPTION
**Pull request by dmcavoy**
We had an issue where a levelbuilder was in a Spanish locale on levelbuilder. They went in and edited a script and when it saved it changed the name and description of the script to the Spanish version. This change got deployed and users for confused why their course was now in Spanish when their locale was not.

To prevent this we are going to limit editing of scripts, levels, etc on levelbuilder to only the en-US locale. If you try to perform an editing type action in another locale you will get redirected to the homepage and see a flash notice (see below).

Screen Shot 2020-05-26 at 5 24 45 PM

This change will impact actions in the following controllers:
blocks_controller.rb
courses_controller.rb
datasets_controller.rb
level_assets_controller.rb
level_starter_assets_controller.rb
levels_controller.rb
libraries_controller.rb
script_levels_controller.rb
scripts_controller.rb
shared_blockly_functions_controller.rb
videos_controller.rb

## Links

- [jira](https://codedotorg.atlassian.net/browse/PLAT-164)

## Testing story

I added a test to make sure you get redirected if you try to edit a script or a level in another locale besides en-US.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/application_controller.rb [hunk 1]
```diff
@@ -208,7 +208,13 @@ def set_locale_cookie(locale)
     cookies[:language_] = {value: locale, domain: :all, expires: 10.years.from_now}
   end
 
+  def require_english_in_levelbuilder_mode
+    redirect_to '/', flash: {alert: 'Editing on levelbuilder is only supported in English (en-US locale).'} unless locale == :'en-US'
+  end
+
   def require_levelbuilder_mode
+    require_english_in_levelbuilder_mode
```
Comment (davidsbailey) [inline dashboard/app/controllers/application_controller.rb:216]: thank you for pushing for this more general approach, Dani! I think this is quite likely to save us from another surprise unwanted translation somewhere besides the script and level edit pages.

**Review by davidsbailey — APPROVED** [review]
Great solution Dani! In addition to the test coverage you added, we also have at least one UI test covering the `require_levelbuilder_mode_or_test_env` pathway on the script edit page.

**Review by davidsbailey — APPROVED** [review]
new UI tests look good!

File: dashboard/app/controllers/application_controller.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/app/controllers/application_controller.rb:216]: +1000

**Comment by Hamms** [comment]
This looks great!

**Pull request merged**